src/infra/core/src/sync_service_impl.rs

```diff
@@ -145,12 +144,21 @@ impl SyncServiceImpl {
                     }
                     Err(GetDatasetError::NotFound(_)) if create_if_not_exists => {
                         let alias = local_ref.alias().unwrap().clone();
-                        let repo_writer = self.dataset_repo_writer.clone();
+                        let create_dataset_use_case = self.create_dataset_use_case.clone();
```

Comment:
Normally service layer wouldn't invoke a higher level component like use case.
But here, we are rather missing use case layer for push/pull operations.

Reply:
Yes, I had my doubts too, but came to the conclusion that it's acceptable here (in this case)